waspc/src/Wasp/AppSpec.hs

```diff
@@ -75,7 +75,8 @@ data AppSpec = AppSpec
     configFiles :: [ConfigFileRelocator],
     -- | Connection URL for a database used during development. If provided, generated app will
     -- make sure to use it when run in development mode.
-    devDatabaseUrl :: Maybe String
+    devDatabaseUrl :: Maybe String,
+    isCustomViteConfigUsed :: Bool
```

Comment:
Since this can be determined based on `externalClientFiles`, I would not have additional field here, because it is introducing redundancy and opportunity for potential desync.

Piece of knowledge we are introducing in this PR is that there can be a vite config file defined by user at specific path(s) and we will use it.
So ideally we want a single place to encode that.

I see two ways to go about it:
1. Let `Wasp.Project` part of `Wasp` have the knowledge of what exact path is, and it will then store that path into AppSpec under `customViteConfigFilePath :: Maybe FilePath` -> ok you should use StrongPath that is relative to project source dir root, not FilePath, but you get the idea. That means that in `Wasp.Project` (or somewhere beneath it) we will have logic that searches for that file and then fills in this field in AppSpec, and from there on, AppSpec caries the info about the path and that is it.
2. Or, we can have AppSpec not have a field at all, but instead have a function called `isCustomViteConfigUsed` that takes app spec and returns Bool but doing the check through the client files. You will end up with the same type signature but it will not be field but a getter that calculates stuff on the fly. But that means that AppSpec has the knowledge of the path of the vite config file in that function, so how do other parts of code learn about that path? I guess then we add a constant somewhere in AppSpec module or submodule, which contains that path to vite config file, and that is used both by `isCustomViteConfigUsed` and also others later (the vite template file).

Of these two, I think I actually like the first one better. We could also check how we do it for other files, where do we encode their paths, but this feels right to me.

Reply:
Now that I think about it, it makes sense for me to calculate the Vite config file path e.g. `Maybe Path' Rel r File f`and keep in the AppSpec. The field could be called `customViteConfigPath`

1. If it's there, use it in the template
2. If other parts of the code need it, it is also there for them to consume

It feels okay to keep it in the AppSpec to me since it's calculated at the same time as the list of external client files. Calculating it on the fly is okay to me, but it feels redundant to calculate the same thing that won't change multiple times.

Reply:
Sounds good! Calculating multiple times sounds a bit wasteful, but it is not really expensive and it would work quite well in this case I think -> however, I also like better that this logic, this knowledge of knowing where the path to custom vite config is, is present not in AppSpec, but in the logic that creates AppSpec. So great!